File: 프로그래머스/lv2/42883. 큰 수 만들기/큰 수 만들기.cpp
#include <string>
#include <vector>

using namespace std;

string solution(string number, int k) {
    string answer = "";
    for (;k > 0;)
    {
        int temp = k;
        for (int i = 0; i < number.length() - 1; i++)
        {
            if (number[i] < number[i + 1])
            {
                number.erase(i, 1);
                k--;
                break;
            }
        }
        if (temp == k)
            break ;
    }
    for (; k > 0; k--)
        number.pop_back();
    answer = number;
    return answer;
}

File: 프로그래머스/lv2/42883. 큰 수 만들기/큰 수 만들기.js
function solution(number, k) {
	const answer = [];

	for (let i = 0; i < number.length; i++) {
		if (answer.length !== 0 && k > 0 && answer[answer.length - 1] < number[i]) {
			answer.pop();
			k--;
			i--;
			continue;
		}
		answer.push(number[i]);
	}
	while (k > 0) {
		answer.pop();
		k--;
	}
	return answer.join('');
}